### ISSUE TITLE
Specifying Different Supplementary Views Per Section

### ISSUE THREAD

**Issue by RyanCodes** [DenHeadless/DTCollectionViewManager#25]
How can I specify that I want a certain supplementary view to show up in a certain section of the collectionView? 

Maybe i am missing where in the docs I can do this

e.x
Section 1 = HeaderViewOne
Section 2 = HeaderViewTwo
.
.
.
Section n = HeaderViewN

**Comment by DenHeadless** [comment]
This depends on whether you want them to have different models or not. If data models for different sections is different, so will be the views, all you need is register all your views, and just set supplementaries on a storage.

If data types of your models is the same, then you can customize mappings using `ViewModelMappingCustomizing` protocol like so:

```swift
extension ViewController: ViewModelMappingCustomizing {
    func viewModelMapping(from candidates: [ViewModelMapping], for model: Any) -> ViewModelMapping? {
        // some kind of model conversion
       if model.kind == .one {
             return candidates.findOneMapping() // hypothethical method
       } else {
           // ...
       }
    }
}
```

It's a little cumbersome, but you'll be able to specify supplementary views for your models. I think in future releases it's possible that framework will allow overriding datasource methods to create such views without data models, but it's only theory.

Unfortunately, framework also does not have "bindings by section", not for cells, not for supplementary views, so you won't be able to specify mappings for section directly, as described in the issue. It's an interesting idea, and i heard it several times already, but did not have time to properly implement it.

Do first two solutions solve your problem?

**Comment by DenHeadless** [comment]
Hey @RyanCodes!

I've released 6.0.0-beta.1 release of `DTCollectionViewManager`, which includes conditional mappings feature, that allows you to specify mappings per section, like so:

```swift
manager.register(SectionHeader1.self) { mapping in mapping.condition = .section(0) }
manager.register(SectionHeader2.self) { mapping in mapping.condition = .section(1) }
```

Hope this helps!
